styx-service-common/src/main/java/com/spotify/styx/storage/DatastoreStorage.java

```diff
@@ -582,6 +592,25 @@ static Entity runStateToEntity(KeyFactory keyFactory, WorkflowInstance wfi, RunS
     return entity.build();
   }
 
+  @VisibleForTesting
+  static Entity workflowToEntity(Workflow workflow, WorkflowState state, Optional<Entity> existing, Key key)
```

Comment:
It's nice to see this code abstracted away in this method. the store method family became way simpler and easier to read.